Update Demo (#2)
jbw976/demo-xfn: kcl/README.md

message: created resource "vpc-1:VPC"
severity: SEVERITY_NORMAL
step: render-with-kcl
```

## Deploying this Composition Resources to AWS

Now that we have tested locally, we can create these resources in AWS.

In the next few steps, we'll:

* Install Crossplane
* Install the AWS Provider and set up Authentication
* Install the KCL Function
* Apply the CompositeResourceDefinition (XRD) and Composition
* Create your Claim which will trigger the Composition to create the Resources

### Installing Crossplane

Install Crossplane via the [Getting Started](https://docs.crossplane.io/v1.15/getting-started/provider-aws/) guide. For this example a minimum Crossplane version of 1.14.x is required.

### Installing the EC2 Provider

> Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.1.1>. A copy of the manifest is in this directory.

Reviewer: ```suggestion
Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.0>. A copy of the manifest is in this directory.
```

The major Cloud providers (AWS, Azure, GCP) are broken up in to "family" providers in order to reduce CRDs installed on the cluster, which is why there is a separate provider for the EC2 related resources.

```shell
kubectl apply -f provider.yaml
```

> Verify that your Provider was installed Correctly. You should see that the `upbound-provider-family-aws` provider was automatically installed. This provider supplies the common `ProviderConfig` CRD.

Reviewer: ```suggestion
Verify that your Provider was installed correctly. You should see that the `upbound-provider-family-aws` provider was automatically installed. This provider supplies the common `ProviderConfig` CRD.
```

```shell
❯ kubectl get provider.pkg
NAME                      INSTALLED   HEALTHY   PACKAGE                                          AGE
upbound-provider-aws-ec2   True        True      xpkg.upbound.io/upbound/provider-aws-ec2:v1.2.0   9m

> upbound-provider-family-aws              True        True      xpkg.upbound.io/upbound/provider-family-aws:v1.2.0              10m

Reviewer: is it easy to make these columns aligned? if not then don't worry 😉

kind: ProviderConfig
metadata:
  name: default
spec:
  credentials:
    source: Secret
    secretRef:
      namespace: crossplane-system
      name: aws-secret
      key: creds
```

### Installing the Functions

Functions are also installed via a Crossplane package. The KCL function can be found at <https://marketplace.upbound.io/functions/crossplane-contrib/function-kcl/v0.2.0>. We will also be using the [auto-ready](https://marketplace.upbound.io/functions/crossplane-contrib/function-auto-ready/v0.2.1) function to ensure resource ready status is propagated to the Composite Resource.

```yaml
apiVersion: pkg.crossplane.io/v1beta1
kind: Function
metadata:
  name: function-kcl
spec:
  package: xpkg.upbound.io/crossplane-contrib/function-kcl:v0.2.0
```

```console
❯ kubectl apply -f functions.yaml
function.pkg.crossplane.io/function-kcl created
function.pkg.crossplane.io/function-auto-ready created
```

Ensure the KCL functions are healthy:

```console
❯ kubectl get -f functions.yaml 
NAME           INSTALLED   HEALTHY   PACKAGE                                                  AGE
function-kcl   True        True      xpkg.upbound.io/crossplane-contrib/function-kcl:v0.2.0   114s

> function-auto-ready   True        True      xpkg.upbound.io/crossplane-contrib/function-auto-ready:v0.2.1   118s

Reviewer: same here for these columns to be aligned

message: created resource "vpc-1:VPC"
severity: SEVERITY_NORMAL
step: render-with-kcl
```

## Deploying this Composition Resources to AWS

Now that we have tested locally, we can create these resources in AWS.

In the next few steps, we'll:

* Install Crossplane
* Install the AWS Provider and set up Authentication
* Install the KCL Function
* Apply the CompositeResourceDefinition (XRD) and Composition
* Create your Claim which will trigger the Composition to create the Resources

### Installing Crossplane

Install Crossplane via the [Getting Started](https://docs.crossplane.io/v1.15/getting-started/provider-aws/) guide. For this example a minimum Crossplane version of 1.14.x is required.

### Installing the EC2 Provider

> Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.1>. A copy of the manifest is in this directory.

Reviewer: ```suggestion
Once Crossplane is installed, install the AWS Provider from the manifest at <https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v1.2.0>. A copy of the manifest is in this directory.
```

Reviewer: the link as is gives me a 404
